controllers/object_controls.go

```diff
@@ -773,6 +783,48 @@ func applyCommonDaemonsetConfig(obj *appsv1.DaemonSet, config *gpuv1.ClusterPoli
 	return nil
 }
 
+// Transform the host-root volume if required for the Daemonset passed as input
+func transformHostRootVolume(obj *appsv1.DaemonSet, hostRoot string) {
+	if hostRoot == "" || hostRoot == "/" {
+		return
+	}
+
+	containsHostRootVolume := false
+	for index, volume := range obj.Spec.Template.Spec.Volumes {
+		if volume.Name == "host-root" {
+			obj.Spec.Template.Spec.Volumes[index].HostPath.Path = hostRoot
```

Comment:
Since we're not modifying `Volumes` here, should this not just be:
```suggestion
			volume.HostPath.Path = hostRoot
```
at which point we don't need an explicit variable.

Reply:
Updated as suggested.